src/build/__init__.py

```diff
@@ -17,10 +17,35 @@
 import zipfile
 
 from collections import OrderedDict
-from typing import AbstractSet, Any, Callable, Dict, Iterator, Mapping, Optional, Sequence, Set, Tuple, Type, Union
+from typing import (
+    AbstractSet,
+    Any,
+    Callable,
+    Dict,
+    Iterator,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    Union,
+)
 
 import pep517.wrappers
-import toml
+
+
+TOMLDecodeError: Type[Exception]
+toml_loads: Callable[[str], MutableMapping[str, Any]]
+
+
+try:
+    from tomli import TOMLDecodeError
+    from tomli import loads as toml_loads
+except ModuleNotFoundError:  # pragma: no cover
+    from toml import TomlDecodeError as TOMLDecodeError  # type: ignore
+    from toml import loads as toml_loads  # type: ignore
```

Comment:
Eh, I'm not that eager on this. I'd prefer to just drop this. If we must keep on this we should at least validate that build works with toml and not tomli by adding a test env for this :shrug:

Reply:
It does seem a bit pointless considering pep517 has switched to tomli for Python 3.6 and above.

Reply:
This is still relevant for bootstraping. Building with older version is common if it makes things eaier.

The code won't affect any use that installed the package the normal way, just people provisioning environments.

Reply:
But considering pep517 switched to tomli, and we depend on tomli wouldn't people need to provision tomli anyway? If they only provide toml, pep517 will break...

Reply:
Sure, but currently it's easier to build an older version on pep517 and then update it instead of building tomli directly when bootstraping.

Reply:
I see we were missing this piece of information. Perhaps worth adding this as comment in the catch 👍

Reply:
Can't they also bootstrap an older version of build and then update it, what's the difference?

Reply:
> I see we were missing this piece of information. Perhaps worth adding this as comment in the catch +1

I improved the comment in `setup.cfg`, let me know if you want me to clarify it more, and if so how/what exactly.

> Can't they also bootstrap an older version of build and then update it, what's the difference?

They won't get bugfixes or improvements. As long as we have a usable version of pep517, things should be good, I find it much more likely that bugs will happen on our part.
I'd also like for pep517 to fallback to toml, which should be pretty straightforward as they have the code already because they still support Python < 3.6.